libindy-crypto/tests/cl.rs

```diff
@@ -0,0 +1,700 @@
+extern crate indy_crypto;
+
+use indy_crypto::cl::Predicate;
+use indy_crypto::cl::issuer::Issuer;
+use indy_crypto::cl::prover::Prover;
+use indy_crypto::cl::verifier::Verifier;
+
+pub const PROVER_ID: &'static str = "CnEDk9HrMnmiHXEV1WFgbVCRteYnPqsJwrTdcZaNhFVW";
+
+mod test {
+    use super::*;
+    use indy_crypto::ffi::ErrorCode;
+    use indy_crypto::errors::ToErrorCode;
+
+    #[test]
+    fn anoncreds_demo() {
+        // 1. Issuer creates claim schema
+        let mut claim_schema_builder = Issuer::new_claim_schema_builder().unwrap();
+        claim_schema_builder.add_attr("name").unwrap();
+        claim_schema_builder.add_attr("sex").unwrap();
+        claim_schema_builder.add_attr("age").unwrap();
+        claim_schema_builder.add_attr("height").unwrap();
+        let claim_schema = claim_schema_builder.finalize().unwrap();
+
+        // 2. Issuer creates keys
+        let (issuer_pub_key, issuer_priv_key) = Issuer::new_keys(&claim_schema, false).unwrap();
+
+        // 3. Prover creates master secret
+        let master_secret = Prover::new_master_secret().unwrap();
+
+        // 4. Prover blinds master secret
+        let (blinded_ms, master_secret_blinding_data) = Prover::blind_master_secret(&issuer_pub_key, &master_secret).unwrap();
+
+        // 5. Issuer creates claim values
+        let mut claim_values_builder = Issuer::new_claim_values_builder().unwrap();
+        claim_values_builder.add_value("name", "1139481716457488690172217916278103335").unwrap();
+        claim_values_builder.add_value("sex", "5944657099558967239210949258394887428692050081607692519917050011144233115103").unwrap();
+        claim_values_builder.add_value("age", "28").unwrap();
+        claim_values_builder.add_value("height", "175").unwrap();
+        let claim_values = claim_values_builder.finalize().unwrap();
+
+        // 6. Issuer signs claim values
+        let mut claim_signature = Issuer::sign_claim(PROVER_ID,
+                                                     &blinded_ms,
+                                                     &claim_values,
+                                                     &issuer_pub_key, &issuer_priv_key,
+                                                     None,
+                                                     None,
+                                                     None).unwrap();
+
+        // 7. Prover processes claim signature
+        Prover::process_claim_signature(&mut claim_signature, &master_secret_blinding_data, &issuer_pub_key, None).unwrap();
+
+        // 8. Verifier create sub proof request
+        let mut sub_proof_request_builder = Verifier::new_sub_proof_request().unwrap();
```

Comment:
Please create a demo test where we have multiple claims (from different issuers) and multiple sub-proofs for different claims.